CMUG Sea Surface Salinity dataset and diagnostic (#1832)
ESMValGroup/ESMValTool: doc/sphinx/source/recipes/recipe_sea_surface_salinity.rst

.. _recipes_sea_surface_salinity:

Sea Surface Salinity Evaluation
===============================

Overview
--------

This recipe compares the regional means of sea surface salinity with a
reference dataset (ESACCI-SEA-SURFACE-SALINITY v1 or v2 by default).

> To do this, the recipe generate plots for the timeseries of each region and

Reviewer: ```suggestion
To do this, the recipe generates plots for the timeseries of each region and
```

> a radar plot showing the correlation of dataset and reference timeseries for

Reviewer: ```suggestion
a radar plot showing the correlation between dataset and reference timeseries for
```

> each region during the time they both exists.

Reviewer: ```suggestion
each region during the time they both exist.
```

Reviewer: what do you mean by the "the time they both exist"?

Preprocessor requirements:

> ^^^^^^^^^^^^^^^^^^^^^^^^^^

Reviewer: please use Markdown syntax ie replace "^^^" with "---"

Reviewer 2: Changed as suggested.

The recipe is created in a way that should make possible (although is not

> tested) to use it for other variables and datasets, even for more that one at

Reviewer: ```suggestion
tested) to use it for other variables and datasets, even for more than one at
```

Reviewer: if it hasn't been tested please don't mention it; or, to the very least, mention it at the bottom via a `NOTE`

a time. The diagnostic only expects variables with dimensions `time` and `depth_id`
and it does not assume any other constraint.

> It ist herefore mandatory to keep the `extract_shape` preprocessor for more than

Reviewer: ```suggestion
It is therefore mandatory to keep the `extract_shape` preprocessor for more than
```

- Gulf of California
- Gulf of Finland
- Gulf of Guinea
- Gulf of Mexico
- Gulf of Oman
- Gulf of Riga
- Gulf of St. Lawrence
- Gulf of Suez
- Gulf of Thailand
- Gulf of Tomini
- Halmahera Sea
- Hudson Bay
- Hudson Strait
- Indian Ocean
- Inner Seas off the West Coast of Scotland
- Ionian Sea
- Irish Sea and St. George's Channel
- Japan Sea
- Java Sea
- Kara Sea
- Kattegat
- Labrador Sea
- Laccadive Sea
- Laptev Sea
- Ligurian Sea
- Lincoln Sea
- Makassar Strait
- Malacca Strait
- Mediterranean Sea - Eastern Basin
- Mediterranean Sea - Western Basin
- Molukka Sea
- Mozambique Channel
- North Atlantic Ocean
- North Pacific Ocean
- North Sea
- Norwegian Sea
- Persian Gulf
- Philippine Sea
- Red Sea
- Rio de La Plata
- Savu Sea
- Sea of Azov
- Sea of Marmara
- Sea of Okhotsk
- Seto Naikai or Inland Sea
- Singapore Strait
- Skagerrak
- Solomon Sea
- South Atlantic Ocean
- South China Sea
- South Pacific Ocean
- Southern Ocean
- Strait of Gibraltar
- Sulu Sea
- Tasman Sea
- The Coastal Waters of Southeast Alaska and British Columbia
- The Northwestern Passages
- Timor Sea
- Tyrrhenian Sea
- White Sea
- Yellow Sea

Available recipes and diagnostics
---------------------------------

> Recipes are stored in recipes/

Reviewer: ```suggestion
Recipes that are stored in recipes/
```

    * recipe_sea_surface_salinity.yml

Diagnostics are stored in diag_scripts/sea_surface_salinity/

    * compare_salinity.py: plot timeseries for each region and generate radar
      plot.

User settings in recipe
-----------------------

#. compare_salinity.py

   *Required settings for script*

   none

   *Optional settings for script*

   none

   *Required settings for variables*

   * ref_model: name of reference data set

   *Optional settings for variables*

   none

Variables
---------

* sos (ocean, monthly, time latitude longitude)

Observations and reformat scripts
---------------------------------

* ESACCI-SEA-SURFACE-SALINITY (sos)

References
----------

> * Please, contact authors

Reviewer: at least one reference would be good to have, a paper or an OBS dataset?

Reviewer 2: Good point, I added a reference for the ESACCI dataset to the docu.

.. _recipes_sea_surface_salinity:

> Sea Surface Salinity Evaluation

Reviewer 2: I think it would be good to have a section in the documentation on:

(1) how to obtain the shapefile needed
(2) in which directory to put the shapefile
(3) available regions (or at least examples) for analysis
(4) required / recommended preprocessor settings (extract_shape)

===============================

Overview
--------

This recipe compares the regional means of sea surface salinity with a
reference dataset (ESACCI-SEA-SURFACE-SALINITY v1 or v2 by default).
To do this, the recipe generate plots for the timeseries of each region and
a radar plot showing the correlation of dataset and reference timeseries for
each region during the time they both exists.

The recipe is created in a way that should make possible (although is not
tested) to use it for other variables and datasets, even for more that one at
a time.

Available recipes and diagnostics
---------------------------------

Recipes are stored in recipes/

>     * recipe_sea_surface_salinity

Reviewer 2: ```suggestion
    * recipe_sea_surface_salinity.yml
```

Diagnostics are stored in diag_scripts/sea_surface_salinity/

    * compare_salinity.py: plot timeseries for each region and generate radar
      plot.

User settings in recipe
-----------------------

#. compare_salinity.py

   *Required settings for script*

   none

   *Optional settings for script*

   none

   *Required settings for variables*

   * ref_model: name of reference data set

   *Optional settings for variables*

   none

   *Required settings (scripts)*

   none

   *Optional settings (scripts)*

Variables
---------

> * sos (ocean, monthly mean, time depth_id)

Reviewer 2: ```suggestion
* sos (ocean, monthly, longitude, latitude, time)
```

Author: This should refer to the original data or the dimensions expected by the diagnostic after the preprocessor?

Reviewer 2: So far, this ususally referred to the dimensions expected by the recipe (recipe as provided in the official release). I guess we should think about how to handle this in the future and make it more clear to what this actually refers. Or maybe simply get rid of this section? What do people think?
